core/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -284,8 +285,9 @@ private void executeSearch(SearchTask task, SearchTimeProvider timeProvider, Sea
         for (int i = 0; i < indices.length; i++) {
             concreteIndices[i] = indices[i].getName();
         }
+        Map<String, Long> nodeSearchCounts = searchTransportService.getPendingRequests(SearchAction.NAME);
```

Comment:
this is puzzeling to me, why are you useing the `SearchAction.NAME` it's the action that acts as a coordinator and we don't necessarily run this from within the cluster so the counts are expected to be 0 for almost all nodes? I wonder if that should be the shard actions here?